nexus/src/app/background/tasks/fm_sitrep_load.rs

```diff
@@ -0,0 +1,332 @@
+            }
+            Some(old) if current_version.version < old.version => {
+                warn!(
+                    log,
+                    "current sitrep version v{} is less than the previously \
+                     loaded version v{}; ignoring it",
+                    current_version.version,
+                    old.version,
+                );
+                return Status::Error(format!(
+                    "current sitrep version v{} is less than the previously \
+                     loaded version v{}; ignoring it",
+                    current_version.version, old.version,
+                ));
+            }
+            Some(SitrepVersion { version, id, .. })
+                if version == current_version.version
+                    && id != current_version.id =>
+            {
+                // Well, this is weird! Entries in the `sitrep_version` table
+                // should not change IDs once they are created, that seems like
+                // a bug. Nonetheless, we will load the new UUID, but we should
+                // say something about this, as it's a bit odd.
+                warn!(
+                    log,
+                    "sitrep ID associated with the current version in the \
+                     database has changed; this is not supposed to happen!";
+                     "current_id" => ?current_version.id,
+                );
+            }
+            _ => (),
+        }
+
+        let sitrep = match self
+            .datastore
+            .fm_sitrep_read(opctx, current_version.id)
+            .await
+        {
+            Ok(sitrep) => sitrep,
+            Err(err) => {
+                let err = InlineErrorChain::new(&err);
+                error!(
+                    log,
+                    "failed to load current sitrep";
+                    "current_id" => ?current_version.id,
+                    "current_version" => ?current_version.version,
+                    &err
+                );
+                return Status::Error(format!(
+                    "failed to read current sitrep {:?} (v{}): {err}",
+                    current_version.id, current_version.version
+                ));
+            }
+        };
+
+        let sitrep = Arc::new((current_version.clone(), sitrep));
+        self.tx.send_modify(|s| {
+            *s = Some(sitrep);
+        });
+
+        Status::Loaded { version: current_version, time_loaded }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::app::background::BackgroundTask;
+    use nexus_db_queries::db::pub_test_utils::TestDatabase;
+    use nexus_types::fm::SitrepMetadata;
+    use omicron_test_utils::dev;
+    use omicron_uuid_kinds::CollectionUuid;
+    use omicron_uuid_kinds::OmicronZoneUuid;
+    use omicron_uuid_kinds::SitrepUuid;
+
+    #[tokio::test]
+    async fn test_load_sitreps() {
+        let logctx = dev::test_setup_log("test_inventory_loader");
```

Comment:
(I think I made this comment on the commit that added this, but not sure if github actually correlates that with the PR. Why would it? Sigh).


Wrong test name?